generate/template/astronomy.kt

```diff
@@ -3247,10 +3233,9 @@ private fun getPlutoSegment(tt: Double): List<BodyGravCalc>? {
         return null     // Don't bother calculating a segment. Let the caller crawl backward/forward to this time
 
     val segIndex = clampIndex((tt - plutoStateTable[0].tt) / PLUTO_TIME_STEP, PLUTO_NUM_STATES-1)
-    synchronized (plutoCache) {
-        var list = plutoCache.get(segIndex)
-        if (list == null) {
-            val seg = ArrayList<BodyGravCalc>()
+    return synchronized(plutoCache) {
+        plutoCache.getOrPut(segIndex) {
```

Comment:
This getOrPut is one nice thing provided in Kotlin for such tasks https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/get-or-put.html